WordPressAuthenticatorTests/Mocks/ModalViewControllerPresentingSpy.swift

```diff
@@ -0,0 +1,15 @@
+@testable import WordPressAuthenticator
+
+
+protocol ModalViewControllerPresenting {
+    func present(_ viewControllerToPresent: UIViewController, animated: Bool, completion: (() -> Void)?)
+}
+
+extension UIViewController: ModalViewControllerPresenting {}
```

Comment:
This code should go in the production target, and the methods we test passing the spy should be updated to expect `ModalViewControllerPresenting` instead of `UIViewController`.

Right now, the spy works because it subclasses `UIViewController` not because of the protocol itself. Subclassing `UIViewController` is exactly what we want to avoid by adding the protocol as an extra abstraction layer in between, because when one starts to add `UIViewController`s to the view hierarchy in the tests, things can sometimes go wrong due to window and view management.

Reply:
Makes sense.  Thanks :)